### PR TITLE
LIIKUNTA-344 | feat(sports): filter venues by target groups' ontology tree IDs

### PR DESCRIPTION
**Pull request by karisal-anders**
### feat(sports): filter venues by target groups' ontology tree IDs

target group filtering:
 - data mappings:
   - remove TARGET_GROUPS_TO_UNIFIED_SEARCH_TARGET_GROUPS
   - add TARGET_GROUPS_TO_ONTOLOGY_TREE_IDS with target group mappings
	 - add ADAPTED_GROUPS_ONTOLOGY_TREE_IDS
	 - add ADULTS_ONTOLOGY_TREE_IDS
	 - add CHILDREN_AND_FAMILIES_ONTOLOGY_TREE_IDS
	 - add PARTNERS_ONTOLOGY_TREE_IDS
	 - add SENIORS_ONTOLOGY_TREE_IDS
	 - add YOUTH_ONTOLOGY_TREE_IDS
 - user interface:
   - show target group filter by default if not turned off by
	 NEXT_PUBLIC_SHOW_TARGET_GROUP_FILTER environment variable
 - queries & types:
   - add ontologyTreeIdsOrSet2 to unifiedSearchList.query.ts
   - update generated types
 - VenueSearchAdapter:
   - add ontologyTreeIdsOrSet2 output parameter
	 - map target groups to ontology tree IDs into this output parameter
   - remove targetGroups output parameter
   - refactor ontology tree ID related mapping functions
 - tests:
   - fix related tests
   - add tests for target groups' ontology tree ID & keyword lists
	 having no duplicates

also:
 - fix linting warnings/errors reported by "yarn g:lint-fix"
   - remove unused variables and imports
 - fix EventInfo tests
   - search for "5–15-vuotiaat" instead of "5-15 -vuotiaat"
   - search for "5+-vuotiaat" instead of "5+ -vuotiaat"
 - rename targetGroupData to TARGET_GROUP_DATA

refs LIIKUNTA-344

### Closes

**[LIIKUNTA-344](https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-344)**

## Additional notes

[LIIKUNTA-344]: https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-344?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr513.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr513.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr513.dev.hel.ninja :laughing::tada::tada::tada:

File: apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts [hunk 1]
```diff
@@ -33,10 +32,10 @@ class VenueSearchAdapter implements CombinedSearchAdapter<VenueSearchParams> {
   language: VenueSearchParams['language'];
   q: VenueSearchParams['q'];
   ontologyTreeIds: VenueSearchParams['ontologyWordIds'];
+  ontologyTreeIdsOrSet2: VenueSearchParams['ontologyTreeIdsOrSet2'];
```
Comment (nikomakela) [inline apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts:35]: Just out of curiousity, why is it called "...set2"?

**Review by nikomakela — APPROVED** [review]

File: apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts (hunk 1 above)
Comment (karisal-anders) [reply apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts:35]: For similarity with the LinkedEvents related keywordOrSet<X> names, although here I didn't rename ontologyTreeIds to ontologyTreeIdsOrSet1

**Pull request merged**